src/psyclone/tests/psyGen_test.py

```diff
@@ -663,6 +664,96 @@ def test_codedkern_lower_to_language_level(monkeypatch):
+                           api="lfric")
+    psy = PSyFactory("lfric", distributed_memory=False).create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    kern = schedule.children[0].loop_body[0]
+    assert isinstance(kern, CodedKern)
+
+    # Ensure everything is as expected before renaming
+    kern_schedule = kern.get_kernel_schedule()
+    assert kern_schedule.name == "testkern_code"
+
+    container = kern_schedule.ancestor(Container)
+    assert isinstance(container, Container)
+    assert container.name == "testkern_mod"
+    container_table = container.symbol_table
+    testkern_type = container_table.lookup("testkern_type")
+    assert isinstance(testkern_type.datatype, StructureType)
+    assert len(testkern_type.datatype.procedure_components) == 1
+    assert "code" in testkern_type.datatype.procedure_components
+    assert testkern_type.datatype.procedure_components["code"].name == "code"
+    assert isinstance(testkern_type.datatype.procedure_components["code"]
+                      .datatype,
+                      UnsupportedFortranType)
+
+    # Rename the kernel in the PSyIR tree
+    kern._rename_psyir(suffix="_new")
+    kern_schedule = kern.get_kernel_schedule()
+    assert kern_schedule.name == "testkern_new_code"
+    container = kern_schedule.ancestor(Container)
+    assert container.name == "testkern_new_mod"
+    container_table = container.symbol_table
+    testkern_type = container_table.lookup("testkern_type")
+    assert isinstance(testkern_type.datatype.procedure_components["code"]
+                      .datatype,
+                      UnsupportedFortranType)
+    # In this case string substitution is applied in the .declaration string
+    assert ("code => testkern_new_code"
+            in testkern_type.datatype.procedure_components["code"].datatype
+            .declaration)
+
+    # Make the procedure component of the structure type be of a supported type
+    _, invoke_info = parse(os.path.join(BASE_PATH, "1_single_invoke.f90"),
+                           api="lfric")
+    psy = PSyFactory("lfric", distributed_memory=False).create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    kern = schedule.children[0].loop_body[0]
+    assert isinstance(kern, CodedKern)
+    kern_schedule = kern.get_kernel_schedule()
+    container = kern_schedule.ancestor(Container)
+    container_table = container.symbol_table
+    testkern_type = container_table.lookup("testkern_type")
+    procedure_component = testkern_type.datatype.procedure_components["code"]
+    assert isinstance(procedure_component.datatype, UnsupportedFortranType)
+    # Replace the procedure component
+    new_procedure_component = StructureType.ComponentType(
+        procedure_component.name,
+        UnresolvedType(),
+        procedure_component.visibility,
+        # This initial_value part should be renamed
+        Reference(RoutineSymbol("testkern_code")))
```

Comment:
Shouldn't this `RoutineSymbol` be in the parent SymbolTable though?

Reply:
Indeed! Replaced with a lookup